core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournal.java

```diff
@@ -436,14 +437,14 @@ public String toString() {
 
   @Override
   public synchronized void close() throws IOException {
-    if (mWriter != null) {
-      mWriter.close();
-      mWriter = null;
-    }
     if (mAsyncWriter != null) {
       mAsyncWriter.close();
       mAsyncWriter = null;
     }
+    if (mWriter != null) {
```

Comment:
is the order of close b/w async and sync writer important? if so can you leave a comment

Reply:
Yes, otherwise pending items on async queue could get lost. Added comments.